google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -113,12 +113,18 @@ public CancellableRunnable createDeployRunner(
       AppEngineDeploymentConfiguration deploymentConfiguration,
       DeploymentOperationCallback callback) {
 
-    if (source.getFile() == null
-        || !(source instanceof AppEngineDeployable)) {
-      callback.errorOccurred(GctBundle.message("appengine.deployment.source.not.found.error"));
+    if (!(source instanceof AppEngineDeployable)) {
+      callback.errorOccurred(GctBundle.message("appengine.deployment.invalid.source.error"));
       throw new RuntimeException("Invalid deployment source selected for deployment");
```

Comment:
I guess this shouldn't happen so this case really involves an error on our part?